### ISSUE TITLE
Add Person.nr_ids to persons/creators/narrators fields

### ISSUE THREAD

**Issue by gjost** [denshoproject/ddr-local#314]
Enable ddr-cmdln/ddr-local to handle NRIDs in persons/creators/narrators fields

Look at e.g. http://ddrlocal.local/ui/entity/ddr-densho-16-2/
- more structured data in creators field
- less structured data in persons
- can you put more structured data in persons?
- can you add fields to creators data?

This
```
    "persons": [
        "Domoto, Sally",
        "Fujii, Kenji \"Kizi\""
    ],
```
should get automatically converted to this
```
    "persons": [
        {"namepart" : "Domoto, Sally"},
        {"namepart": "Fujii, Kenji \"Kizi\""}
    ]
```
You should then be able to add NRIDs like this
```
    "persons": [
        {"namepart" : "Domoto, Sally"},
        {"namepart": "Fujii, Kenji \"Kizi\"", "nr_id": "88922/nr014m435", "provisional":"true"}
    ]
```
sara.beckman: (Archival Team) In the names field, regardless of which style of cataloging you choose to complete, names must be written as Last name, First name “Nickname” (Maiden Name). A semi-colon must separate people. At a minimum, a last name and either a first name or nickname (without quotes) must be included in the field. If a first name or nickname is unknown, do not include in the names field, include in the description field only. (i.e. Yoda, Tats or Yoda, Tatsu are both acceptable, but just Mr. Yoda is not. If you find out that “Tats” is Tatsu’s nickname, change his name in every entry of the name field to Yoda, Tatsu “Tats”- goal is to have one name for every unique person in the database)

**Comment by gjost** [comment]
*Saving* `nr_id` to the `creators` field seems to work. This
```
nr_id:88922/nr014m435|namepart:Domoto, Sally|role:photographer;
nr_id:88922/nr014m435|namepart:Fujii, Kenji|role:writer
```
gets written to `entity.json` as this
```
{
    "creators": [
        {
            "namepart": "Domoto, Sally",
            "nr_id": "88922/nr014m435",
            "role": "photographer"
        },
        {
            "namepart": "Fujii, Kenji",
            "nr_id": "88922/nr014m435",
            "role": "writer"
        }
    ]
},
```
But *loading* breaks it: the `nr_id` values are screened out the next time the Entity is edited, looking like this in the field:
```
namepart:Domoto, Sally|role:photographer; namepart:Fujii, Kenji|role:writer
```

**Comment by gjost** [comment]
`nr_id` is present after `Entity` is loaded in the `edit` view:
```
creators | [{'namepart': 'Domoto, Sally',   'nr_id': '88922/nr014m435',   'role': 'photographer'},  {'namepart': 'Fujii, Kenji', 'nr_id': '88922/nr014m435', 'role': 'writer'}]
```

**Comment by gjost** [comment]
Improvements to `rolepeople` parsing:

- [ ] 1. Parse creators,person during form validation, before sending to Celery

- [ ] 2. Print and accept spaces around formatting characters
Currently: ```namepart:Domoto, Sally|nr_id:88922/nr014m435|role:photographer;```
Desired: ```namepart:Domoto, Sally | nr_id:88922/nr014m435 | role:photographer;```

- [ ] 3. Print names on separate lines by default
Treat newlines and semicolons interchangeably.

- [ ] 4. Reorder field data in forms. `nr_id` first for alignment, readability.
nr_id:88922/nr014m435 | namepart:Domoto, Sally | role:photographer;
nr_id:88922/nr014m435 | namepart:Fujii, Kenji | role:writer

**Comment by gjost** [comment]
Merged `314-person-nrids` into `develop` for `ddr-cmdln`, `ddr-local`.  Edits for `ddr-defs` remain on `314-person-nrids` branch.

**Comment by gjost** [comment]
Reopening this issue.  It's time to make `persons` a structured field like `creators`.

**Comment by gjost** [comment]
Testing procedure
https://gist.github.com/gjost/b63f0e94487285aec775bea275f69a73.js

**Comment by gjost** [comment]
Merged in to `master` branch and released as `v5.8.1`.
